mozilla/bedrock
# Add Playwright test suite (Fixes #14191)

## One-line summary

This PR adds a new suite of integration tests written using https://playwright.dev/. 

## Significant changes and points to review

Right now, this is only run manually and is not (yet) part of our CI deployment process. Once this merges, the next step is to start to run this against a deployment for a while to evaluate it for stability.

Aims of this new test suite:

1. It's written in JavaScript as opposed to Python, so will hopefully be more familiar for front-end developers to maintain.
2. It should be significantly easier to get running locally (everything is managed via an NPM install).
3. It hopefully adds more valuable tests (e.g. asserting auto file downloads, testing older UA strings) compared to the Selenium test suite (which had quite a number of old, superfluous tests for less critical pages).
4. Runs tests in all three major browser engines (Gecko, Chromium, Webkit).

If we like this better than Selenium and it's stable enough, we can look at potentially retiring the old test suite.

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/14191

## Testing

1. Start bedrock locally via `npm start`.
2. In a new terminal tab navigate to the playwright directory: `cd tests/playwright/`.
3. Install playwright via `npm install` (I opted for a separate `package.json` as the browser binaries are quite large).
5. Download latest version of browser binaries: `npm run install-deps`.

You should then be able to run Playwright tests (see the `./tests/playwright/README.md` file for instructions).

- [ ] Whole test suite should pass when running`npx playwright test`.
